Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -891,6 +893,7 @@ template <class TreeBuilder> TreeExpression Parser<LexerType>::parseVariableDecl
 
         failIfTrue(match(PRIVATENAME), "Cannot use a private name to declare a variable");
         if (matchSpecIdentifier()) {
+            failIfTrue(isArgumentsToken() && !findClosestFunctionScopeUntilStaticBlock(), "Class static block cannot use 'arguments' as an identifier");
```

Comment:
Let's just check ident with `arguments` directly. We should not call `match(IDENT)` again.

Reply:
There is a chance that it's not `IDENT` type and `matchSpecIdentifier` returns true.
```
    ALWAYS_INLINE bool matchSpecIdentifier()
    {
        return match(IDENT) || isAllowedIdentifierLet(m_token) || isAllowedIdentifierYield(m_token) || isPossiblyEscapedAwait(m_token);
    }
```

Reply:
I think if it is spec identifier, then the token ident exists, so we don't need to check match(IDENT) here anyway